static/app/views/organizationIntegrations/sentryAppExternalForm.tsx

```diff
@@ -0,0 +1,323 @@
+import {Component} from 'react';
+import {createFilter} from 'react-select';
+import debounce from 'lodash/debounce';
+
+import {addErrorMessage} from 'app/actionCreators/indicator';
+import {Client} from 'app/api';
+import {t} from 'app/locale';
+import {SentryAppInstallation} from 'app/types';
+import {replaceAtArrayIndex} from 'app/utils/replaceAtArrayIndex';
+import withApi from 'app/utils/withApi';
+import FieldFromConfig from 'app/views/settings/components/forms/fieldFromConfig';
+import Form from 'app/views/settings/components/forms/form';
+import FormModel from 'app/views/settings/components/forms/model';
+import {Field, FieldValue} from 'app/views/settings/components/forms/type';
+
+// 0 is a valid choice but empty string, undefined, and null are not
+const hasValue = value => !!value || value === 0;
+
+export type FieldFromSchema = Omit<Field, 'choices' | 'type'> & {
+  type: 'select' | 'textarea' | 'text';
+  default?: 'issue.title' | 'issue.description';
+  uri?: string;
+  depends_on?: string[];
+  choices?: Array<[any, string]>;
+  async?: boolean;
+};
+
+export type Config = {
+  uri: string;
+  required_fields?: FieldFromSchema[];
+  optional_fields?: FieldFromSchema[];
+};
+
+// only need required_fields and optional_fields
+type State = Omit<Config, 'uri'> & {
+  optionsByField: Map<string, Array<{label: string; value: any}>>;
+};
+
+type Props = {
+  api: Client;
+  sentryAppInstallation: SentryAppInstallation;
+  appName: string;
+  config: Config;
+  action: 'create' | 'link';
+  element: 'issue-link' | 'alert-rule-action';
+  extraFields?: {[key: string]: any};
+  extraRequestBody?: {[key: string]: any};
+  onSubmitSuccess: (...params: any[]) => void;
+  getFieldDefault?: (field: FieldFromSchema) => string;
```

Comment:
These new fields allow for compatibility with the alert rule actions UI, while still allowing the Issue Linking to pass in its Fields (for the ExternalIssueStore), and Request Body (projectId) which it's API needs.

Reply:
getFieldDefault is a function unique to the parent which renders this form. For IssueLinking it auto-fills `issue.title` and `issue.description`. This can be extended for alert rule actions by letting the parent handle getting field defaults.